lib/src/cancelable_operation.dart

```diff
@@ -189,7 +189,7 @@ class CancelableCompleter<T> {
     final future = value as Future<T>;
     if (_isCanceled) {
       // Make sure errors from [value] aren't top-leveled.
-      future.catchError((_) {});
+      future..then<void>((_) {}).onError((_, __) {});
```

Comment:
I'd do: `future.then<void>((_) {}, onError: (_, __) {})` here. It's guaranteed type-safe because of the `void`, so no need to create two futures where one will do.

Reply:
The SDK now has an `.ignore()` extension method on futures which ignores errors and does not create a new future that you need to handle.